app/src/lib/helpers/linux.ts

```diff
@@ -0,0 +1,85 @@
+import { join } from 'path'
+import { pathExists as pathExistsInternal } from 'fs-extra'
+import {
+  ChildProcess,
+  spawn,
+  SpawnOptionsWithoutStdio,
+  SpawnOptions,
+} from 'child_process'
+
+export function isFlatpakBuild() {
+  return __LINUX__ && process.env.FLATPAK_HOST === '1'
+}
+
+/**
+ * Convert an executable path to be relative to the flatpak host
+ *
+ * @param path a path to an executable relative to the root of the filesystem
+ *
+ */
+function convertToFlatpakPath(path: string) {
+  return join('/var/run/host', path)
+}
+
+/**
+ * Checks the file path on disk exists before attempting to launch a specific shell
+ *
+ * @param path
+ *
+ * @returns `true` if the path can be resolved, or `false` otherwise
+ */
+export async function pathExists(path: string): Promise<boolean> {
+  if (isFlatpakBuild()) {
+    path = convertToFlatpakPath(path)
+  }
+
+  try {
+    return await pathExistsInternal(path)
+  } catch {
+    return false
+  }
+}
+
+/**
+ * Spawn a particular shell in a way that works for Flatpak-based usage
+ *
+ * @param path path to shell, relative to the root of the filesystem
+ * @param args arguments to provide to the shell
+ * @param options additional options to provide to spawn function
+ *
+ * @returns a child process to observe and monitor
+ */
+export function spawnShell(
+  path: string,
+  args: string[],
+  options?: SpawnOptionsWithoutStdio
+): ChildProcess {
+  if (isFlatpakBuild()) {
+    return spawn('flatpak-spawn', ['--host', path, ...args], options)
```

Comment:
🤔 I think this is the convention for mapping the original command (and parameters) to work with `flatpak-spawn`, but now that it's all centralized we can tweak this to suit.